histoflux/src/app_context.rs

```diff
@@ -76,6 +75,40 @@ impl SqsProducer {
         })
     }
 
+    /// This function returns a [`HistoFluxCursor`] from the database. If the
+    /// cursor does not exist, it creates a new one and returns it.
+    async fn get_or_create_cursor(
+        pg_pool: &PgPool,
+        env: &Env,
+    ) -> Result<HistoFluxCursor, HistoFluxError> {
+        let cursor = HistoFluxCursor::find_by_environment(pg_pool, &env.environment).await?;
+        if let Some(cursor) = cursor {
+            Ok(cursor)
+        } else {
+            NewHistoFluxCursor::builder()
+                .last_processed_id(0)
+                .environment(env.environment.clone())
+                .paused(false)
+                .queue_url(Self::get_queue_url(env))
+                .build()
+                .insert(pg_pool)
+                .await
+        }
+    }
+
+    /// This function returns the queue URL based on the environment.
+    fn get_queue_url(env: &Env) -> String {
+        if let Some(_localstack_url) = &env.localstack_url {
+            "http://sqs.us-east-1.localhost.localstack.cloud:4566/000000000000/raw_logs.fifo"
+                .to_string()
+        } else {
+            format!(
+                "https://sqs.us-west-2.amazonaws.com/064662847354/prod-eks-prod-{}-raw-logs.fifo",
+                env.environment
+            )
+        }
+    }
```

Comment:
It looks like this could be simplified. Maybe introduce
```
SQS_URL_TEMPLATE=https://sqs.us-west-2.amazonaws.com/064662847354/prod-eks-prod-{}-raw-logs.fifo
```